pkg/plugin/datasource.go

```diff
@@ -192,7 +193,20 @@ func (d *Datasource) query(_ context.Context, _ backend.PluginContext, dataQuery
 				backoffDuration = MAX_BACKOFF_DURATION
 			}
 			backend.Logger.Debug("query not finished, delaying/backing off", "backoffDuration", backoffDuration.String())
-			time.Sleep(backoffDuration)
+			select {
+			case <-ctx.Done():
+				err := ctx.Err()
+				if errors.Is(err, context.Canceled) {
+					err := d.SearchAPI.CancelQuery(jobId)
+					if err != nil {
+						backend.Logger.Warn("failed to cancel query", "jobId", jobId, "err", err)
+					}
+					backend.Logger.Warn("query canceled", "jobId", jobId, "err", err)
```

Comment:
2-for-1 on the nits (lucky you!)

1. This probably shouldn't get logged if we failed to cancel the query, so maybe pop this in an `else` block.
2. Maybe `Info` for this?  It's kinda expected in a lot of "normal" scenarios where the user navigates away from the dashboard, closes the browser, etc. right?